sca/src/main/kotlin/Report.kt

```diff
@@ -0,0 +1,37 @@
+package org.example
+
+import org.example.rule.RuleFailure
+import org.example.rule.RuleFailures
+import org.example.rule.RuleResult
+import org.example.rule.RuleSuccess
+import position.Position
+
+class Report(
+    val ruleResults: List<RuleResult>
+){
+    fun print(){
+        if (ruleResults.isEmpty()){
+            println("Static code analysis successful!")
+            return
+        }
+        for (ruleResult in ruleResults) {
+            when(ruleResult){
+                is RuleSuccess -> {}
+                is RuleFailure -> printRuleFailure(ruleResult)
+                is RuleFailures -> printFailures(ruleResult)
```

Comment:
Why does it need to be handled separated ? Can't we consider `RuleFailure` as a special case of `RuleFailures`

Reply:
You mean use the same "printFailures" function?

Reply:
Yes you are right, I could change the whole thing to just use RuleFailures.